src/Autofac/Core/Activators/Reflection/IConstructorSelectorWithEarlyBinding.cs

```diff
@@ -0,0 +1,28 @@
+﻿// Copyright (c) Autofac Project. All rights reserved.
+// Licensed under the MIT License. See LICENSE in the project root for license information.
+
+namespace Autofac.Core.Activators.Reflection;
+
+/// <summary>
+/// Defines an interface that indicates a constructor selector can attempt to
+/// determine the correct constructor early, as the container is built,
+/// without needing to understand the set of parameters in each resolve.
+/// </summary>
+public interface IConstructorSelectorWithEarlyBinding : IConstructorSelector
```

Comment:
```suggestion
public interface IConstructorSelectorWithEarlyBinding
```

It seems like the way .NET Core did it, the "extension feature interface" [didn't inherit the interface of the thing it was extending](https://github.com/dotnet/runtime/blob/main/src/libraries/Microsoft.Extensions.DependencyInjection.Abstractions/src/ISupportRequiredService.cs). I don't know which way is right, I can see pros and cons to each.

Admittedly, you can't really _just_ implement `IConstructorSelectorWithEarlyBinding`, but then, you also can't _just_ implement `ISupportRequiredService`, either. They only make sense together.

Reply:
My feeling here is that in order to provide a constructor selector to Autofac at registration time, it _must_ implement `IConstructorSelector`; that's the type of the property in `ReflectionActivatorData`.  It feels like it would be confusing if someone created an implementation of `IConstructorSelectorWithEarlyBinding`, and then just couldn't provide it to Autofac?

Reply:
Agreed. I just didn't know if I was missing a use case somewhere, like some future composability system like the `PartCatalog` or `HttpContext.Features` sort of thing in ASP.NET Core. I'm fine with it either way, we can leave it.